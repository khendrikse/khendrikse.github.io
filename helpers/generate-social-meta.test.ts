import generateSocialMeta, { trimContent } from './generate-social-meta';

describe('trimContent', () => {
  it('trims a string to the right size', () => {
    expect(trimContent('A long string that can be smaller', 15)).toBe(
      'A long strin...'
    );
  });
});

describe('generateSocialMeta', () => {
  it('generates social meta tags using default fields', () => {
    const result = generateSocialMeta();
    expect(result).toMatchSnapshot();
  });

  it('generates social meta tags using the passed object', () => {
    const input = {
      twitterCardType: 'summary',
      twitterSite: 'A site',
      twitterCreator: 'A creator',
      description: 'A description',
      title: 'A title',
      image: 'image-alt.png',
      imageAlt: 'An image alt',
      type: 'website',
      url: 'fake-url'
    };
    const result = generateSocialMeta(input);
    expect(result).toMatchSnapshot();
  });
});
