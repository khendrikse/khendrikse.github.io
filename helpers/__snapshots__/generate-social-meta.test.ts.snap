// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateSocialMeta generates social meta tags using default fields 1`] = `
Array [
  <React.Fragment>
    <title>
      Karin Hendrikse
    </title>
    <meta
      content="Karin Hendrikse"
      name="twitter:title"
    />
    <meta
      content="Karin Hendrikse"
      property="og:title"
    />
  </React.Fragment>,
  <React.Fragment>
    <meta
      content="A personal playground and portfolio."
      name="description"
    />
    <meta
      content="A personal playground and portfolio."
      name="twitter:description"
    />
    <meta
      content="A personal playground and portfolio."
      property="og:description"
    />
  </React.Fragment>,
  <meta
    content="@k_henhey"
    name="twitter:site"
  />,
  <meta
    content="@k_henhey"
    name="twitter:creator"
  />,
  <meta
    content="summary"
    name="twitter:card"
  />,
  <meta
    content="website"
    property="og:type"
  />,
]
`;

exports[`generateSocialMeta generates social meta tags using the passed object 1`] = `
Array [
  <React.Fragment>
    <title>
      A title
    </title>
    <meta
      content="A title"
      name="twitter:title"
    />
    <meta
      content="A title"
      property="og:title"
    />
  </React.Fragment>,
  <React.Fragment>
    <meta
      content="A description"
      name="description"
    />
    <meta
      content="A description"
      name="twitter:description"
    />
    <meta
      content="A description"
      property="og:description"
    />
  </React.Fragment>,
  <meta
    content="A site"
    name="twitter:site"
  />,
  <meta
    content="A creator"
    name="twitter:creator"
  />,
  <meta
    content="summary"
    name="twitter:card"
  />,
  <meta
    content="website"
    property="og:type"
  />,
  <meta
    content="https://velvety-tartufo-f1de54.netlify.app/fake-url"
    property="og:url"
  />,
  <React.Fragment>
    <meta
      content="image-alt.png"
      name="twitter:image"
    />
    <meta
      content="image-alt.png"
      property="og:image"
    />
  </React.Fragment>,
  <meta
    content="An image alt"
    name="twitter:image:alt"
  />,
]
`;
